### ISSUE TITLE
custom labels for pie charts

### ISSUE THREAD

**Issue by raffomania** [chartist-js/chartist#216]
Other chart types seem to support custom labels specified by the data object. Pie charts don't have this possibility - is that intentional?

**Comment by gionkunz** [comment]
Hey there. What do you mean by custom labels? You can pass optional series name, series class and meta data using the object notation of the data array in Pie charts like you do in the other charts:

``` javascript
var chart = new Chartist.Pie('.ct-chart', {
  labels: ['One', 'Two', 'Three'],
  series: [{
    name: 'My Custom Name 1',
    className: 'my-custom-class-1',
    data: 1
  }, {
    name: 'My Custom Name 2',
    className: 'my-custom-class-2',
    data: 2
  }, {
    name: 'My Custom Name 3',
    className: 'my-custom-class-3',
    data: 3
  }]
});
```

**Comment by raffomania** [comment]
Hey @gionkunz,
thanks! That's exactly what I needed. The documentation for Chartist.Pie doesn't mention this possibility, though - is there a chance you could add a sentence explaining that?
Also, I couldn't find any information regarding the `name` property of the series objects, what is their purpose?

**Comment by gionkunz** [comment]
The `name` property will be written into a custom attribute `ct:name` which can be used in DOM manipulation after the chart is created. If you got time, please make a pull request to add an example to the documentation here: https://github.com/gionkunz/chartist-js/blob/develop/src/scripts/charts/pie.js#L273-L286

**Comment by joshiashish23** [comment]
I can not generate pie chart in Chartist.js 0.7.3 version but it works in Chartist.js 0.5.0.  I want to generate horizontal bar chart and pie chart which is support in newer version not in older version. Can any one help me out in this? Bar chart is working fine with following data but pie chart is not working.

  var CategoryValueTitleArray = [];
        var CategoryValuePointsArray = [];
        @foreach (var item in Model.LeadScoringExplicitValues)
        {

```
    <text>
    CategoryValueTitleArray.push('@item.Value');
    CategoryValuePointsArray.push('@item.Point');
    </text>        
    }
```

  var data = {
            labels: CategoryValueTitleArray,
            series: CategoryValuePointsArray
        };

```
    var options = {
        labelInterpolationFnc: function (value) {
            return value[0]
        }
    };

    var responsiveOptions = [
      ['screen and (min-width: 640px)', {
          chartPadding: 30,
          labelOffset: 50,
          labelDirection: 'explode',
          labelInterpolationFnc: function (value) {
              return value;
          }
      }],
      ['screen and (min-width: 1024px)', {
          labelOffset: 40,
          chartPadding: 20
      }]
    ];

    new Chartist.Pie('#explicit-value-piecanvas', data, options, responsiveOptions);
```

**Comment by raffomania** [comment]
@gionkunz I'll do that in the next few days. Thanks a lot!
@joshiashish23, your problem doesn't seem to be relevant to this issue. Also, I'm not sure if what you are describing is a bug - maybe try creating a question on stackoverflow?

**Comment by gionkunz** [comment]
fixed with 62a80d2
